base/generator_eltype.jl

```diff
@@ -0,0 +1,28 @@
+function eltype(::Type{Generator{A, typeof(identity)}}) where {A}
+    eltype(A)
+end
+
+function IteratorEltype(::Type{Generator{A, typeof(identity)}}) where {A}
+    IteratorEltype(A)
+end
+
+function eltype(::Type{Generator{A, Fix1{typeof(getindex), B}}} where {A}) where {B}
+    if B <: Type
+        # a user may overload `getindex(user_type)` to return a non-`Vector` `AbstractVector`
+        Any
+    else
+        eltype(B)
+    end
+end
```

Comment:
I'm not sure `eltype(B)` is correct, unless eltype(A) matches keytype(B) or something?
```julia
julia> it = Iterators.map(Base.Fix1(getindex, [10, 20, 30]), [[1 2], [3 2 1]])
Base.Generator{Vector{Matrix{Int64}}, Base.Fix1{typeof(getindex), Vector{Int64}}}(Base.Fix1{typeof(getindex), Vector{Int64}}(getindex, [10, 20, 30]), [[1 2], [3 2 1]])

julia> it isa Base.Generator{A, Base.Fix1{typeof(getindex), B}} where {A, B}
true

julia> collect(it)
2-element Vector{Matrix{Int64}}:
 [10 20]
 [30 20 10]

julia> eltype(ans)
Matrix{Int64} (alias for Array{Int64, 2})
```

Reply:
Should be OK now.